ceph-aio-no-vagrant.sh

```diff
@@ -95,37 +114,53 @@ function cp_var {
 }
 
 function populate_vars {
-  sed -i "s/#osd_auto_discovery: false/osd_auto_discovery: true/" group_vars/osds.yml
-  sed -i "s/#journal_collocation: false/journal_collocation: true/" group_vars/osds.yml
-  sed -i "s/#pool_default_size: 3/pool_default_size: 2/" group_vars/all.yml
-  sed -i "s/#monitor_address: 0.0.0.0/monitor_address: ${IP}/" group_vars/all.yml
-  sed -i "s/#journal_size: 0/journal_size: 100/" group_vars/all.yml
-  sed -i "s|#public_network: 0.0.0.0\/0|public_network: ${SUBNET}|" group_vars/all.yml
-  sed -i "s/#common_single_host_mode: true/common_single_host_mode: true/" group_vars/all.yml
+  sed -i "s/[#]*osd_auto_discovery: .*/osd_auto_discovery: true/" group_vars/osds.yml
+  sed -i "s/[#]*journal_collocation: .*/journal_collocation: true/" group_vars/osds.yml
+  sed -i "s/[#]*monitor_address: .*/monitor_address: ${IP}/" group_vars/all.yml
+  sed -i "s/[#]*journal_size: .*/journal_size: 100/" group_vars/all.yml
+  sed -i "s|[#]*public_network: .*|public_network: ${SUBNET}|" group_vars/all.yml
+  sed -i "s/[#]*common_single_host_mode: .*/common_single_host_mode: true/" group_vars/all.yml
+  grep -q '^[#]*osd_pool_default_size: .*' group_vars/all.yml \
+    && sed -i "s/^[#]*osd_pool_default_size: .*/osd_pool_default_size: $CEPH_POOL_DEFAULT_SIZE/" group_vars/all.yml \
+    || echo "osd_pool_default_size: $CEPH_POOL_DEFAULT_SIZE" >> group_vars/all.yml
+  grep -q '^[#]*osd_pool_default_min_size: .*' group_vars/all.yml \
+    && sed -i "s/^[#]*osd_pool_default_min_size: .*/osd_pool_default_min_size: $CEPH_POOL_DEFAULT_SIZE/" group_vars/all.yml \
+    || echo "osd_pool_default_min_size: $CEPH_POOL_DEFAULT_SIZE" >> group_vars/all.yml
+  grep -q '^[#]*mon_pg_warn_max_per_osd: .*' group_vars/all.yml \
+    && sed -i "s/^[#]*mon_pg_warn_max_per_osd: .*/mon_pg_warn_max_per_osd: 0/" group_vars/all.yml \
+    || echo "mon_pg_warn_max_per_osd: 0" >> group_vars/all.yml
   if [[ ${SOURCE} == 'stable' ]]; then
-    sed -i "s/#ceph_stable: false/ceph_stable: true/" group_vars/all.yml
+    sed -i "s/[#]*ceph_stable: .*/ceph_stable: true/" group_vars/all.yml
   else
-    sed -i "s/#ceph_dev: false/ceph_dev: true/" group_vars/all.yml
-    sed -i "s|#ceph_dev_branch: master|ceph_dev_branch: ${BRANCH}|" group_vars/all.yml
+    sed -i "s/[#]*ceph_dev: .*/ceph_dev: true/" group_vars/all.yml
+    sed -i "s|[#]*ceph_dev_branch: .*|ceph_dev_branch: ${BRANCH}|" group_vars/all.yml
   fi
 }
 
 function create_inventory {
-  cat > /etc/ansible/hosts <<EOF
+  cat > hosts <<EOF
 [mons]
 localhost
 [osds]
 localhost
+EOF
+  if [ "$INSTALL_MDS" = true ] ; then
+    cat >> hosts <<EOF
```

Comment:
I don't think this is idempotent, in the condition can you also check for an existing section? Therefore we don't add this section N times while re-running the script.